### PR TITLE
build providers: set hostname for lxd

### PR DESCRIPTION
**Pull request by cjp256**
The hostname is being written to /etc/hostname, but the kernel
is not being informed of the update.  Use hostname -F to set it
after writing.

Signed-off-by: Chris Patterson <chris.patterson@canonical.com>

- [ ] Have you followed the [guidelines for contributing](https://github.com/snapcore/snapcraft/blob/master/CONTRIBUTING.md)?
- [ ] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [ ] Have you successfully run `./runtests.sh static`?
- [ ] Have you successfully run `./runtests.sh tests/unit`?

-----

### CODE REVIEW

**Comment by cjp256** [comment]
Reported at: https://forum.snapcraft.io/t/snapcraft-under-lxd-very-slow-sudo-hostname-problem/24308/2

File: tests/unit/build_providers/lxd/test_lxd.py [hunk 1]
```diff
@@ -246,6 +245,7 @@ def test_create(self):
         ]
 
         assert check_output_calls == [
+            ["hostname", "-F", "/etc/hostname"],
```
Comment (cmatsuoka) [inline tests/unit/build_providers/lxd/test_lxd.py:248]: I think we should have a test that verifies if the hostname is actually being set inside the container.

Comment (sergiusens) [reply tests/unit/build_providers/lxd/test_lxd.py:248]: We can use the already existing --use-lxd test and have an override-<something> perform a check

Comment (cjp256) [reply tests/unit/build_providers/lxd/test_lxd.py:248]: The existing test uses make-hello, so I added a new spread test.

**Review by sergiusens — APPROVED** [review]

**Review by cmatsuoka — APPROVED** [review]

**Pull request merged**